modules/core/src/main/java/org/apache/ignite/internal/processors/metric/impl/MetricUtils.java

```diff
@@ -83,6 +83,32 @@ public static boolean compareAndSet(LongMetricImpl m, long expect, long update)
         return LongMetricImpl.updater.compareAndSet(m, expect, update);
     }
 
+    /**
+     * Update metrics value only if current value if less then {@code update}.
+     *
+     * @param m Metric to update.
+     * @param update New value.
+     */
+    public static void setIfLess(LongMetricImpl m, long update) {
+        long v = m.value();
+
+        while (v > update && !LongMetricImpl.updater.compareAndSet(m, v, update))
+            v = m.value();
+    }
+
+    /**
+     * Update metrics value only if current value if greater then {@code update}.
+     *
+     * @param m Metric to update.
+     * @param update New value.
+     */
+    public static void setIfGreater(LongMetricImpl m, long update) {
+        long v = m.value();
+
+        while (v < update && !LongMetricImpl.updater.compareAndSet(m, v, update))
```

Comment:
I think it's better to use Java 8 approach here with UnarnaryOperator - 

`oldVal.updateAndGet(x -> x < newVal ? newVal : x) == newVal;`

Reply:
Proposed implementation is less efficient then current because `updateAndGet` implementation executes at least one CAS operation(see `AtomicLongFieldUpdater#updateAndGet` sources) event value is not change. And my implementation returns immediately in the case we don't want to update the value.